cmd/compute-domain-kubelet-plugin/device_state.go

```diff
@@ -135,20 +134,31 @@ func (s *DeviceState) Prepare(ctx context.Context, claim *resourceapi.ResourceCl
 
 	claimUID := string(claim.UID)
 
-	checkpoint := newCheckpoint()
-	if err := s.checkpointManager.GetCheckpoint(DriverPluginCheckpointFileBasename, checkpoint); err != nil {
+	checkpoint, err := s.getCheckpoint()
+	if err != nil {
 		return nil, fmt.Errorf("unable to get checkpoint: %w", err)
 	}
 
-	preparedClaim, exists := checkpoint.V1.PreparedClaims[claimUID]
-	if exists {
+	preparedClaim, exists := checkpoint.V2.PreparedClaims[claimUID]
+	if exists && preparedClaim.CheckpointState == ClaimCheckpointStateCompleted {
```

Comment:
Nice.

Add a short code comment here, maybe linking to the github issue that this fixes?

The huge relevance of the new

```
preparedClaim.CheckpointState == ClaimCheckpointStateCompleted
```
may not be obvious to all future readers of this section of code.

Reply:
This doesn't fix anything in particular on its own though.

It's just a byproduct of now having states that we must do this. The thing that fixes the https://github.com/NVIDIA/k8s-dra-driver-gpu/issues/504 issue is in the `Unprepare` when I still call `unprepareDevice()` even if I'm only in the `Started` state.

Reply:
> The thing that fixes the https://github.com/NVIDIA/k8s-dra-driver-gpu/issues/504 issue is in the ....

I get that :). With "maybe linking to the github issue that this fixes?" I meant that this _PR_ fixes, and I quite literally meant with that as an easy way to refer to the complete discussion that motivated the introduction of these states.

Personally, I like adding code comments of the style `// Ref. https://github.com/foo/bar/issues/123` the guide the reader of the code to why something was done. That's a super quick way for guiding future readers of code, w/o having to carefully craft a localized explanation.

I just wanted to explain myself here. Not asking for a change.